halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/GateBoot667ProfileFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2025 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile;
+
+import com.netflix.spinnaker.halyard.config.model.v1.security.Security;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.ServiceSettings;
+import org.springframework.stereotype.Component;
+
+/**
+ * Factory class for creating Gate configuration profiles for versions 6.67.0 and above.
+ *
+ * <p>This class extends {@link GateProfileFactory} and provides specific configurations required
+ * for Gate versions 6.67.0 and later. In these versions, a different set of properties is needed to
+ * enable OAuth2 authentication.
+ *
+ * <p>The factory determines the appropriate security configuration (OAuth2, SAML, LDAP, IAP, X509)
+ * based on the provided {@link Security} settings and constructs the {@link GateConfig}
+ * accordingly.
+ */
+@Component
+public class GateBoot667ProfileFactory extends GateProfileFactory {
```

Comment:
Now that the version has changed, this could use a rename.  Instead of using the version number, let's use a more descriptive name.  Test class too please.